Assets/Scripts/Robot/Robot2DController.cs

```diff
@@ -468,6 +468,75 @@ public void PathAndMoveTo(Vector2Int tile)
             }
         }
 
+        /// <summary>
+        /// Estimates the time of arrival for the robot to reach the specified destination.
+        /// Uses the path from PathAndMoveTo and the robots max speed (RobotConstraints.RelativeMoveSpeed) to calculate the ETA.
+        /// </summary>
+        /// <param name="target">the target that the path should end at.</param>
+        /// <param name="acceptPartialPaths">if <b>true</b>, returns the distance of the path getting the closest to the target, if no full path can be found.</param>
+        /// <param name="beOptimistic">if <b>true</b>, treats unseen tiles as open in the path finding algorithm. Treats unseen tiles as solid otherwise.</param>
+        public int? EstimateTimeToTarget(Vector2Int target, bool acceptPartialPaths = false, bool beOptimistic = false)
+        {
+            // Is Constraints.RelativeMoveSpeed equal to the speed or is this multiplied by the movement force?
+            var distForMaxSpeed = 2.5f;
+            var distance = EstimateDistanceToTarget(target);
+            if (distance == null)
+            {
+                return null;
+            }
+            var dist = distance.Value;
+            var startDist = Math.Min(dist, 2.5f);
+            var startTime = (int)Math.Floor(Math.Pow(CorrectForRelativeMoveSpeed(startDist, Constraints.RelativeMoveSpeed), 0.85));
+            if (dist <= distForMaxSpeed)
+            {
+                return startTime;
+            }
+            else
+            {
+                dist -= distForMaxSpeed;
+                return (int)Math.Ceiling(CorrectForRelativeMoveSpeed(dist, Constraints.RelativeMoveSpeed)) + startTime;
+            }
+
+            static float CorrectForRelativeMoveSpeed(float distance, float relativeMoveSpeed)
+            {
+                return distance * 3.2f / (0.21f + (relativeMoveSpeed / 3.0f));
```

Comment:
Add comments of what the numbers represent

Reply:
Done